### DISCUSSION TITLE
Initial Release

### DISCUSSION THREAD

**Discussion by rwmingis** [rwmingis/InterruptButton#12] [Announcements]
After a fair bit of development, I believe this project is ready for it's first release. 

It is intended that release branch 1.X.X will maintain Async and Sync structure.

There is consideration being made towards using RTOS queues to do the legwork for future releases, and if adopted, will be done in the 2.X.X release branch.

<hr /><em>This discussion was created from the release <a href='https://github.com/rwmingis/InterruptButton/releases/tag/v1.0.0'>Initial Release</a>.</em>

**Comment by vortigont** [comment]
Nice! A good place to start with queues #10

**Reply by rwmingis** [reply]
Wow, yes, the Queues are good.  I have done RTOS tasks before, but need to do a bit of homework on the RTOS Queues.    I am impressed with your Code so far and it does seem to work pretty good.  Shouldn't be too hard to incorporate and maintain the current logic and functionality.  Thanks heaps for the examples.  👍

**Comment by vortigont** [comment]
Sure. I've done some cleanup and now it looks even better. Only minor places left uncertain. If you create some new experimental branch for this, than I can send a PR with my work there. Tailor it to your needs ) For any details we can discuss it under #10 
Cheers!

**Reply by rwmingis** [reply]
Yes!  I have just gone through your code in detail, and you have some great stuff in there, and it reminded me that I probably should eplore updating my formatting at some point too!  :)   

Have managed to update the IB library (offline) to use the RTOS queues based on your example code with minimal effort!  And it works great, I am amazed that this is not a more common method.   I have also created a static class queue (same length as the RTOS queue0 that accepts all button action outputs if the user does a 'setMode(synchronous)' as an alternative to the RTOS queue.  This way they can still use the libarary as a sync library with only a few extra liines of code (using the same bound functions though) as I can see some users wanting to process clicks in their loops to keep things happening as a single flow.  The logging of the clicks is done the same either way which is still better than most libraries i have seen.

Will tidy and upload in the next few days as a new branch.

Cheers and thanks for your great examples.

**Reply by cemik1** [reply]
Thanks you for library. It seems to work in my project. Only drawback - must be installed manually. I use VSCode+PlatformIO and your library is not found trying different search patterns.
